features/user/components/userProfile/ChangePasswordPanel.tsx

```diff
@@ -0,0 +1,189 @@
+'use client'
+
+import { LoadingSpinner } from '@/components/ui/LoadingSpinner'
+import { Button } from '@/components/ui/button'
+import { Card, CardContent, CardHeader } from '@/components/ui/card'
+import {
+	Form,
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import React, { useEffect, useState, useTransition } from 'react'
+import { useForm } from 'react-hook-form'
+import { EditPasswordSchema } from '../../schemas'
+import { z } from 'zod'
+import bcrypt from 'bcryptjs'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { changePassword, getUserById } from '../../actions'
+import { signOut, useSession } from 'next-auth/react'
+import { useToast } from '@/components/ui/use-toast'
+
+type ChangePasswordData = z.infer<typeof EditPasswordSchema>
+
+export const ChangePasswordPanel = () => {
```

Comment:
Imo lepsza nazwa to ChangePasswordTab, zważywszy na to że w UserProfile.tsx używasz Tabów w shadcn'a. To uwaga do wszystkoich plików z /userProfile